packages/spacecat-shared-utils/src/log-wrapper.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/**
+ * Wraps the provided function to enhance the logging functionality by including the
+ * `jobId` in each log statement.
+ *
+ * @param {function} fn - The original function that will be wrapped.
+ * @returns {function(object, object): Promise<Response>} A new function that, when invoked,
+ *  will automatically include the `jobId` in all log statements within the context.
+ */
+export function logWrapper(fn) {
+  return async (message, context) => {
+    const { log } = context;
+
+    if (log) {
+      if (!message || !message.jobId) {
+        log.debug('Missing jobId, hence it will not be included in log messages.');
+        context.contextualLog = log;
```

Comment:
Is `jobLog` no good? It is really job specific. I may have missed a discussion with @bhellema on this.

Reply:
See this discussion: [Slack thread](https://cq-dev.slack.com/archives/C06HYS3AQUF/p1725990669391249?thread_ts=1725989731.295479&cid=C06HYS3AQUF).

I changed the name from jobLog to contextualLog, as we may want to add more properties in the future that provide context, not necessarily related to jobs.